packages/url-loader/src/lib/cloudinary.ts

```diff
@@ -217,25 +208,20 @@ export function constructCloudinaryUrl({ options, config = {}, analytics }: Cons
       options
     });
 
-    const { options: pluginOptions } = results || { options: undefined };
+    const { options: pluginOptions, resize } = results || { options: undefined };
 
-    if ( pluginOptions?.format && options ) {
-      pluginEffects.format = pluginOptions.format;
-    }
+    Object.assign(pluginEffects, pluginOptions);
 
-    if ( pluginOptions?.width && options ) {
-      pluginEffects.resize = {
-        width: pluginOptions?.width
-      };
+    if ( resize ) {
+      pluginResize = resize;
     }
   });
 
   // We want to perform any resizing at the end of the end of the transformation
   // sets to allow consistent use of positioning / sizing, especially responsively
 
-  if ( pluginEffects?.resize && !options.strictTransformations ) {
-    const { width, crop = 'limit' } = pluginEffects.resize;
-    cldAsset.effect(`c_${crop},w_${width}`);
+  if ( pluginResize ) {
+    cldAsset.addTransformation(pluginResize);
```

Comment:
suggestion: maybe we could add this line below 216 like following?

```ts
if ( resize ) {
      pluginResize = resize;

     cldAsset.addTransformation(pluginResize);
}
```

I think in the end it is equal

Reply:
the issue with that is the original `if ( resize ) {` is in the transformation plugins loop, where currently there's only one plugin that's defining a resize, but if there ever is a second, we could potentially be adding multiple resize transformations which im trying to avoid

ex:

if cropping returns `resize: 'w_1000'` and another like fill background for whatever reason returned `resize: w_2000`, they'll both be applied, `w_1000/w_2000` and creates implications, such as, in this example, resizing to 2000 would be upscaling

Reply:
Ah makes sense. Thanks!